From d13ee9c08edb0f8d735390051229823614a0f9ac Mon Sep 17 00:00:00 2001
From: Vishal Sagar <vishal.sagar@xilinx.com>
Date: Thu, 14 Mar 2019 03:55:48 -0700
Subject: [PATCH 0563/1852] v4l: xilinx-demosaic: Make max width and height
 properties mandatory

commit 5b63a0448976cd53f5749112a0f4134b7d577ce2 from
https://github.com/Xilinx/linux-xlnx.git

This patch makes xlnx,max-width and xlnx,max-height dt properties
mandatory. It breaks backward compatibility and has a warning message
for this.

Signed-off-by: Vishal Sagar <vishal.sagar@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/media/platform/xilinx/xilinx-demosaic.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/media/platform/xilinx/xilinx-demosaic.c b/drivers/media/platform/xilinx/xilinx-demosaic.c
index 221366ae0ecb..a519c2c9719b 100644
--- a/drivers/media/platform/xilinx/xilinx-demosaic.c
+++ b/drivers/media/platform/xilinx/xilinx-demosaic.c
@@ -32,8 +32,6 @@
 #define XDEMOSAIC_MIN_WIDTH	(64)
 #define XDEMOSAIC_MAX_WIDTH	(8192)
 #define XDEMOSAIC_DEF_WIDTH	(1280)
-#define XDEMOSAIC_DEF_MAX_WIDTH		(3840)
-#define XDEMOSAIC_DEF_MAX_HEIGHT	(2160)
 
 #define XDEMOSAIC_RESET_DEASSERT	(0)
 #define XDEMOSAIC_RESET_ASSERT		(1)
@@ -266,7 +264,8 @@ static int xdmsc_parse_of(struct xdmsc_dev *xdmsc)
 	rval = of_property_read_u32(node, "xlnx,max-height",
 				    &xdmsc->max_height);
 	if (rval < 0) {
-		xdmsc->max_height = XDEMOSAIC_DEF_MAX_HEIGHT;
+		dev_err(dev, "missing xlnx,max-height property!");
+		return -EINVAL;
 	} else if (xdmsc->max_height > XDEMOSAIC_MAX_HEIGHT ||
 		 xdmsc->max_height < XDEMOSAIC_MIN_HEIGHT) {
 		dev_err(dev, "Invalid height in dt");
@@ -276,7 +275,8 @@ static int xdmsc_parse_of(struct xdmsc_dev *xdmsc)
 	rval = of_property_read_u32(node, "xlnx,max-width",
 				    &xdmsc->max_width);
 	if (rval < 0) {
-		xdmsc->max_width = XDEMOSAIC_DEF_MAX_WIDTH;
+		dev_err(dev, "missing xlnx,max-width property!");
+		return -EINVAL;
 	} else if (xdmsc->max_width > XDEMOSAIC_MAX_WIDTH ||
 		 xdmsc->max_width < XDEMOSAIC_MIN_WIDTH) {
 		dev_err(dev, "Invalid width in dt");
-- 
2.31.1

